doc/developer-guide.rst

```diff
@@ -37,10 +37,10 @@ There is nothing to build except the documentation.
     pytest -v  # to run tests
     package-docs clean; package-docs build  # to build the documentation
 
-.. _lsst.ts.MTDomeTrajectory-contributing:
+.. _lsst.ts.mtdometrajectory-contributing:
 
 Contributing
 ============
 
-``lsst.ts.MTDomeTrajectory`` is developed at https://github.com/lsst-ts/ts_MTDomeTrajectory.
-You can find Jira issues for this module using `labels=ts_MTDomeTrajectory <https://jira.lsstcorp.org/issues/?jql=project%20%3D%20DM%20AND%20labels%20%20%3D%20ts_MTDomeTrajectory>`_.
+``lsst.ts.mtdometrajectory`` is developed at https://github.com/lsst-ts/ts_mtdometrajectory.
+You can find Jira issues for this module using `labels=ts_mtdometrajectory <https://jira.lsstcorp.org/issues/?jql=project%20%3D%20DM%20AND%20labels%20%20%3D%20ts_MTDomeTrajectory>`_.
```

Comment:
I see you didn't rename the Jira label. It is not difficult to do but it of course is up to you if you want to.